src/rgw/services/svc_rados.cc

```diff
@@ -3,307 +3,391 @@
 
 #include "svc_rados.h"
 
-#include "include/rados/librados.hpp"
 #include "common/errno.h"
+#include "common/waiter.h"
 #include "osd/osd_types.h"
 #include "rgw/rgw_tools.h"
 
 #define dout_subsys ceph_subsys_rgw
 
-int RGWSI_RADOS::do_start()
-{
-  int ret = rados.init_with_context(cct);
-  if (ret < 0) {
-    return ret;
+tl::expected<std::int64_t, boost::system::error_code>
+RGWSI_RADOS::open_pool(std::string_view pool, bool create, optional_yield y) {
```

Comment:
note that Sage recently added some mon commands to `rgw_init_ioctx()` in rgw_tools.cc to set pg_num_min/pg_autoscale_bias on created pools that are `mostly_omap`. we'll want to duplicate that here and have some way to pass in `bool mostly_omap`

Reply:
Pulled in.